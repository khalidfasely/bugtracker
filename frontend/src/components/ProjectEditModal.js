import Modal from "react-modal";

import NewProject from "./NewProject";

const ProjectEditModal = ({ editModalOpen, setEditModalOpen, projectItem }) => (
    <Modal
        isOpen={editModalOpen}
        onRequestClose={() => setEditModalOpen(false)}
    >
        <NewProject isEdit={true} projectItemEdit={projectItem} setEditModalOpen={setEditModalOpen} />
    </Modal>
);

export default ProjectEditModal;