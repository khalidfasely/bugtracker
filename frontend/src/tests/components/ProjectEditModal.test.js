import React from "react";
import { render } from "@testing-library/react";

import ProjectEditModal from "../../components/ProjectEditModal";

jest.mock('../../components/NewProject', () => () => <div>NewProject</div>);

let props;

beforeEach(() => {
    props = {
        editModalOpen: true,
        setEditModalOpen: jest.fn()
    };
});

test('Should render ProjectEditModal component correctly', () => {
    const { asFragment } = render(<ProjectEditModal {...props} />);
    expect(asFragment()).toMatchSnapshot();
});