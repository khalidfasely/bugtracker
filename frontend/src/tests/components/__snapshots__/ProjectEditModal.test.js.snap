// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ProjectEditModal component correctly 1`] = `
<DocumentFragment>
  <div>
    NewProject
  </div>
</DocumentFragment>
`;
